rgd/geodata/models/threed/etl.py

```diff
@@ -0,0 +1,80 @@
+import os
+import tempfile
+
+from celery.utils.log import get_task_logger
+from django.conf import settings
+
+from rgd.geodata.models import ChecksumFile
+from rgd.utility import get_or_create_no_commit
+
+from .point_cloud import PointCloudEntry, PointCloudFile
+
+logger = get_task_logger(__name__)
+
+
+def _file_conversion_helper(source, output_field, method, prefix='', extension='', **kwargs):
+    workdir = getattr(settings, 'GEODATA_WORKDIR', None)
+    with tempfile.TemporaryDirectory(dir=workdir) as tmpdir:
+        with source.yield_local_path() as file_path:
+            output_path = os.path.join(tmpdir, prefix + os.path.basename(source.name) + extension)
+            method(str(file_path), str(output_path), **kwargs)
+        with open(output_path, 'rb') as f:
+            output_field.save(os.path.basename(output_path), f)
+
+
+def _save_pyvista(mesh, output_path):
+    import pyvista as pv
+
+    points = pv.PolyData(mesh.points)
+    points.point_arrays.update(mesh.point_arrays)
+    points.save(output_path)
```

Comment:
NOTE: this extracts ONLY the points of the mesh and the point data arrays. We could change this to save any geometry that mesh has as well.

For example, if a user uploads a triangulated mesh, they will only see the point vertices in the front-end viewer.

If we want to always save a PolyData mesh of points and geometry, then we could change this to:

```suggestion
    mesh = mesh.extract_surface()
    mesh.save(output_path)
```